### PR TITLE
Add release workflow

### PR DESCRIPTION
**Pull request by bmangoen**
As describes #290, we  want to use Github Actions for creating releases: 
image

[Create the GitHub repository secrets](https://github.com/istio-ecosystem/sail-operator/settings/secrets/actions):
- [ ] GIT_USER
- [ ] GIT_TOKEN
- [ ] QUAY_USER
- [ ] QUAY_PWD

Resolves https://github.com/istio-ecosystem/sail-operator/issues/290, resolves https://github.com/istio-ecosystem/sail-operator/issues/315

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    - name: Generate bundle metadata
+      run: |
+        make bundle \
+         -e CHANNELS=$CHANNELS
+      env:
+        CHANNELS: ${{ inputs.bundle_channels }}
+
+    - name: Publish bundle in operatorhub.io
+      run: |
+        make bundle-publish \
```
Comment (dgn) [inline .github/ISSUE_TEMPLATE/workflows/release.yaml:59]: We should be able to do the following here to use the username of whoever is triggering the action
``` -e GIT_CONFIG_USER_NAME="${{ github.actor }}"
-e GIT_CONFIG_USER_EMAIL="${{ github.actor_id }}+${{ github.actor }}@users.noreply.github.com"```

Same below for OpenShift

File: hack/operatorhub/publish-bundle.sh [hunk 2]
```diff
@@ -96,6 +100,9 @@ BUNDLE_DIR="${CUR_DIR}"/../../bundle
 mkdir -p "${OPERATORS_DIR}"
 cp -a "${BUNDLE_DIR}"/. "${OPERATORS_DIR}"
 
+skipInDryRun git config user.name "${GIT_CONFIG_USER_NAME}"
```
Comment (dgn) [inline hack/operatorhub/publish-bundle.sh:103]: I think this should be wrapped in an if-clause that checks whether `git config user.name` returns a value - if it's already set, no need to overwrite

**Comment by MaxBab** [comment]
Hi @bmangoen 

Just a small comment.
The path for the github action workflows should be as `.github/workflows/release.yaml` and not under `ISSUE_TEMPLATE` directory - `.github/ISSUE_TEMPLATE/workflows/release.yaml`.
https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions

**Comment by bmangoen** [comment]
> Hi @bmangoen
> 
> Just a small comment. The path for the github action workflows should be as `.github/workflows/release.yaml` and not under `ISSUE_TEMPLATE` directory - `.github/ISSUE_TEMPLATE/workflows/release.yaml`. https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions

Good catch @MaxBab ! Certainly a wrong drag and drop in my IDE

**Review by dgn — APPROVED** [review]

**Comment by sridhargaddam** [comment]
Looking at the logs of some of the recent failures in the job `e2e-kind-olm_sail-operator_main`, the error seems to be always 

```
...
time="2024-09-09T11:44:31Z" level=info msg="Created Subscription: sailoperator-v0-2-0-sub"
time="2024-09-09T11:46:18Z" level=fatal msg="Failed to run bundle: install plan is not available for the subscription sailoperator-v0-2-0-sub: context deadline exceeded\n"
+ cleanup_kind_cluster operator-integration-tests
...
```

**Pull request merged**